clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy

```diff
@@ -405,32 +414,30 @@ class BasicGoogleDeployHandler implements DeployHandler<BasicGoogleDeployDescrip
         .setAutoHealingPolicies(autoHealingPolicy)
 
     if (hasBackendServices && (description?.loadBalancingPolicy || description?.source?.serverGroupName))  {
-      NamedPort namedPort = null
+      List<NamedPort> namedPorts = []
       def sourceGroupName = description?.source?.serverGroupName
+
       // Note: this favors the explicitly specified load balancing policy over the source server group.
       if (sourceGroupName && !description?.loadBalancingPolicy) {
         def sourceServerGroup = googleClusterProvider.getServerGroup(description.accountName, description.source.region, sourceGroupName)
         if (!sourceServerGroup) {
           log.warn("Could not locate source server group ${sourceGroupName} to update named port.")
         }
-        namedPort = new NamedPort(
-            name: GoogleHttpLoadBalancingPolicy.HTTP_PORT_NAME,
-            port: sourceServerGroup?.namedPorts[(GoogleHttpLoadBalancingPolicy.HTTP_PORT_NAME)] ?: GoogleHttpLoadBalancingPolicy.HTTP_DEFAULT_PORT,
-        )
+        sourceServerGroup?.namedPorts?.forEach { name, port ->
+          namedPorts << new NamedPort(name: name, port: port)
```

Comment:
Seems like it would be more consistent to reassign `namedPorts` (as you do below) here rather than mutate the list.